mlflow/server/handlers.py

```diff
@@ -527,24 +527,46 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+_os_alt_seps = list(sep for sep in [os.sep, os.path.altsep] if sep is not None and sep != "/")
+
+
+def validate_path_is_safe(path):
+    """
+    Validates that the specified path is safe to join with a trusted prefix. This is a security
+    measure to prevent path traversal attacks. The implementation is based on
+    `werkzeug.security.safe_join` (https://github.com/pallets/werkzeug/blob/a3005e6acda7246fe0a684c71921bf4882b4ba1c/src/werkzeug/security.py#L110).
+    """
+    if path != "":
+        path = posixpath.normpath(path)
+    if (
+        any(sep in path for sep in _os_alt_seps)
+        or os.path.isabs(path)
+        or path == ".."
+        or path.startswith("../")
```

Comment:
Does this catch paths of the following structure? `my/deceptive/path/../../../../../sensitive/info`?

Reply:
If not, I've tended to find that it works better to start with a mock base path, e.g. `/tmp/testpath` and then do `os.path.join("/tmp/testpath", path)` and make sure that the result is a subdirectory of `/tmp/testpath` or that it's exactly the same directory as `/tmp/testpath`

Reply:
Yes, it does. `posixpath.normpath(path)` converts `my/deceptive/path/../../../../../sensitive/info` to `../../sensitive/info`.

Reply:
Sounds good!